hermeto/core/scm.py

```diff
@@ -209,3 +240,88 @@ def _reset_git_head(repo: Repo, ref: str) -> None:
             "Failed on checking out the Git repository. Please verify the supplied reference "
             f'of "{ref}" is valid.'
         )
+
+
+class GitSupportConfig:
+    """Base configuration class for package manager git support."""
+
+    def __init__(
+        self,
+        supported_schemes: list[str],
+        package_manager: PackageManagerType,
+    ):
+        """Initialize GitSupportConfig.
+
+        Args:
+            supported_schemes: List of supported git URL schemes (e.g. ['git+https', 'git+ssh'])
+            package_manager: The package manager this configuration is for
+        """
+        self.supported_schemes = supported_schemes
+        self.package_manager = package_manager
+
+
+def validate_git_dependency(url: str, config: GitSupportConfig) -> None:
+    """Validate git dependency requirements using package manager configuration.
+
+    Args:
+        url: Git repository URL
+        config: GitSupportConfig object with package manager settings
+
+    Raises:
+        UnsupportedFeature: If git dependencies are not supported or invalid
+    """
+    # Check if git dependencies are supported at all
+    if not config.supported_schemes:
+        raise UnsupportedFeature(
+            f"{APP_NAME} does not support Git dependencies for {config.package_manager}",
+            solution="Use registry-published packages instead of git dependencies",
+        )
+
+    # Parse and validate URL scheme
+    parsed_url = urlparse(url)
+    if parsed_url.scheme not in config.supported_schemes:
+        supported = ", ".join(config.supported_schemes)
+        raise UnsupportedFeature(
+            f"Unsupported VCS scheme for {config.package_manager}: {parsed_url.scheme} "
+            f"(supported: {supported})",
+            solution=f"Use one of the supported git URL schemes: {supported}",
+        )
+
+
+def clone_git_dependency(
+    url: str, ref: str, to_path: Path, config: GitSupportConfig, branch: str | None = None
+) -> None:
+    """Clone a git dependency using the appropriate method for the package manager.
+
+    Args:
+        url: Git repository URL
+        ref: Git reference (full commit hash)
+        to_path: Destination path for cloning
+        config: GitSupportConfig object with package manager settings
+        branch: Optional branch to checkout
+
+    Raises:
+        UnsupportedFeature: If git dependencies are not supported
+        FetchError: If cloning fails
+    """
+
+    validate_git_dependency(url, config)
+
+    log.debug("Cloning git repository from %s for %s", url, config.package_manager)
+
+    try:
+        _clone_git_repo(
+            url=url,
+            to_path=to_path,
+            ref=ref,
+            branch=branch,
+        )
+
+    except Exception as ex:
```

Comment:
Please correct me if I am wrong, but it looks like the only exception that could be raised here is `FetchError`. I'd argue that `Exception` is too broad and could be narrowed down.

Reply:
`_clone_git_repo` essentially converts `Exception` into `FetchError`. While it might make sense to keep `_clone_git_repo` catch the most generic exception possible here it looks like it should always be `FetchError`.